### PR TITLE
feat: add hasSquareCorners prop to bar chart

### PR DESCRIPTION
**Pull request by richiepreece**
## Description

Adding `hasSquareCorners` prop to allow bar charts to render with square tops.

## Related Issue

[Adobe Jira ticket AN-338678](https://jira.corp.adobe.com/browse/AN-338678)

## Motivation and Context

This is needed to allow square tops for an Adobe Product Analytics feature called "Forked Funnel"

## How Has This Been Tested?

Tested in unit tests in this request

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/specBuilder/bar/barUtils.test.ts [hunk 1]
```diff
@@ -230,6 +230,27 @@ describe('barUtils', () => {
 			expect(horizontal?.cornerRadiusBottomRight).toEqual(vertical?.cornerRadiusTopRight);
 			expect(horizontal?.cornerRadiusBottomLeft).toEqual(vertical?.cornerRadiusBottomRight);
 		});
+		test('corner radius should be 0 when the hasSquareCorners prop is true', () => {
+			const squareRadius = getCornerRadiusEncodings({ ...defaultBarProps, hasSquareCorners: true });
+
+			// Square radius should have values of 0
+			expect(squareRadius).toEqual(
+				expect.objectContaining({
```
Comment (marshallpete) [inline src/specBuilder/bar/barUtils.test.ts:238]: I didn't know about `expect.objectContaining`. Very cool. The more you know.

File: src/specBuilder/bar/barUtils.ts [hunk 2]
```diff
@@ -180,8 +180,8 @@ export const getStackedMetricEncodings = (props: BarSpecProps): RectEncodeEntry
 };
 
 export const getCornerRadiusEncodings = (props: BarSpecProps): RectEncodeEntry => {
-	const { type, lineWidth, metric } = props;
-	const value = Math.max(1, CORNER_RADIUS - getLineWidthPixelsFromLineWidth(lineWidth) / 2);
+	const { type, lineWidth, metric, hasSquareCorners } = props;
+	const value = hasSquareCorners ? 0 : Math.max(1, CORNER_RADIUS - getLineWidthPixelsFromLineWidth(lineWidth) / 2);
```
Comment (marshallpete) [inline src/specBuilder/bar/barUtils.ts:184]: Clean and simple

File: src/specBuilder/bar/barUtils.test.ts (hunk 1 above)
Comment (richiepreece) [reply src/specBuilder/bar/barUtils.test.ts:238]: It's a good way to only check about the properties you care about, regardless of the rest of the object shape.

**Comment by marshallpete** [comment]
Can you add one thing? In `specBuilderTypes.ts` ~line 89, there is a type called `BarPropsWithDefaults`. Can you add `hasSquareCorners` to the union of props with defaults? You will probably need to add `hasSquareCorners: false` to a few test files. Run `yarn ts` after updating `specBuilderTypes to know everywhere that you'll need to add that to.

**Comment by marshallpete** [comment]
All of our props that have default values, we update the type so these are no longer optional when used within our code. That way you never have to check if something exists, you know it will. I realize for a boolean prop it's less critical but for consistency it's nice even on boolean props.

**Comment by marshallpete** [comment]
Lastly can you update the prop table in the docs: https://github.com/adobe/react-spectrum-charts/wiki/Bar. Just add in the new prop. Simple update.

**Review by marshallpete — APPROVED** [review]

**Pull request merged**